packages/opencanary/data_stream/events/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,759 @@
+      ignore_failure: true
+  - rename:
+      description: Rename VNC 'VNC Password' field to opencanary.vnc.password
+      tag: Rename VNC 'VNC Password' field to opencanary.vnc.password
+      if: ctx.log?.logger != null && ctx.log.logger == 'LOG_VNC'
+      field: 'opencanary.logdata.VNC Password'
+      target_field: opencanary.tftp.vnc.password
+      ignore_missing: true
+      ignore_failure: true
+  - rename:
+      description: Rename VNC 'VNC Server Challenge' field to opencanary.vnc.server_challenge
+      tag: Rename VNC 'VNC Server Challenge' field to opencanary.vnc.server_challenge
+      if: ctx.log?.logger != null && ctx.log.logger == 'LOG_VNC'
+      field: 'opencanary.logdata.VNC Server Challenge'
+      target_field: opencanary.tftp.vnc.server_challenge
+      ignore_missing: true
+      ignore_failure: true
+
+  ################################
+  ## Parse Event Details to ECS ##
+  ################################
+  - append:
+      description: Append "event.category" according to event details
+      tag: Append "event.category" according to event details
+      field: event.category
+      value: 
+        - network
+        - intrusion_detection
+      allow_duplicates: false
+      ignore_failure: true
+  - date:
+      description: Set "event.created" from "opencanary.utc_time"
+      tag: Set "event.created" from "opencanary.utc_time"
+      field: opencanary.utc_time
+      target_field: event.created
+      formats:
+        - yyyy-MM-dd HH:mm:ss.SSSSSS
+      ignore_failure: true
+  - append:
+      description: Set "event.kind"
+      tag: Set "event.kind"
+      field: event.kind
+      value:
+        - alert
+      allow_duplicates: false
+      ignore_failure: true
+  - set:
+      description: Copy "event.provider" from "log.logger"
+      tag: Copy "event.provider" from "log.logger"
+      field: event.provider
+      copy_from: log.logger
+      ignore_empty_value: true
+      ignore_failure: true
+  - date:
+      description: Set "event.start" from "opencanary.utc_time"
+      tag: Set "event.start" from "opencanary.utc_time"
+      field: opencanary.utc_time
+      target_field: event.start
+      formats:
+        - yyyy-MM-dd HH:mm:ss.SSSSSS
+      ignore_failure: true
+  - set:
+      description: Set "event.type"
+      tag: Set "event.type"
+      field: event.type
+      value:
+        - connection
+
+  ######################################
+  ## Parse Destination Details to ECS ##
+  ######################################
+  - rename:
+      description: Rename dst_port to destination.port
+      tag: Rename dst_port to destination.port
+      field: opencanary.dst_port
+      target_field: destination.port
+      ignore_missing: true
+      ignore_failure: true
+  - convert:
+      description: Convert "destination.port" to integer
+      tag: Convert "destination.port" to integer
+      field: destination.port
+      type: long
+      ignore_missing: true
+      ignore_failure: true
```

If this fails, we'll leave the value in place and this will result in a mapping failure. If it fails, we should remove the field. Suggest instead...
```suggestion
  - convert:
      description: Convert dst_port to destination.port integer.
      tag: convert_opencanary_dst_port_destination_port
      field: opencanary.dst_port
      type: long
      target_field: destination.port
      ignore_missing: true
      ignore_failure: true
```
This means that if it fails to convert, at least it will be retained and the document will map. Strictly, we should provide a field definition for this, but I imagine that it will happen ~0 of the time, and if it does happen, it would map as a keyword which will be fine.

Similar below.

---

This is a shortcoming of elastic-package's understanding of what `keyword` means. There is a work around; in the test config file (opencanary/data_stream/events/_dev/test/pipeline/test-common-config.yml) you can add a definition list of fields that are mapped as keywords in ES but are numeric in the rendered JSON
```
numeric_keyword_fields:
  - opencanary.dst_port
  - opencanary.src_port
```